gridstatus/ercot.py

```diff
@@ -814,6 +823,28 @@ def _finalize_as_price_df(self, doc, pivot=False):
 
         return doc[col_order]
 
+    def _get_historical_document(self, date, verbose=False):
+        r = requests.get("http://www.ercot.com/gridinfo/load/load_hist/")
+        soup = BeautifulSoup(r.content, features="lxml")
+        year_to_document = {}
+        for link in soup.findAll("a"):
+            link_path = link.get("href")
+            if any(extension in link_path for extension in (".xls", ".zip", ".txt")):
+                year = re.findall(r"\d+", link.text)[0]
+                publish_text = link.parent.parent.find("span").text
+                pattern = r"\b(?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s\d{1,2},\s\d{4}\b"  # noqa: E501
+                publish_date = re.findall(pattern, publish_text)[0]
```

Comment:
Assuming the publish date is the one listed on the [site](https://www.ercot.com/gridinfo/load/load_hist)and not the one in the URL. 

For example, 2022 says `Jan 31, 2023 - zip - 1 MB` on the website, but the URL has 2-08-2022 (https://www.ercot.com/files/docs/2022/02/08/Native_Load_2022.zip)